### DISCUSSION TITLE
【建议】添加中文文档

### DISCUSSION THREAD

**Discussion by UpKiry** [spencerwooo/onedrive-vercel-index#352] [Ideas]
这样或许会对新手友好一点 😄

**Comment by Lemonawa** [comment]
+1

**Comment by People-11** [comment]
我的Fork就是将此项目界面尽可能的汉化（仅项目本身，不包含文档）。如果原作者有需要，我可以提供翻译工作。

**Reply by myl7** [reply]
Ref #359

**Reply by spencerwooo** [reply]
@People-11 FYI：

- 文档仓库：https://github.com/spencerwooo/onedrive-vercel-index-docs
- ~~Next.js 国际化：https://nextjs.org/docs/advanced-features/i18n-routing~~ 应该用不到这个，下面的文档就够用了
- Nextra 文档国际化：https://nextra.vercel.app/features/i18n
- 中文文档排版指南：https://github.com/sparanoid/chinese-copywriting-guidelines

edit - 我是不是理解错了，我以为你也要翻译文档，要是我理解错了就忽略我吧

**Reply by People-11** [reply]
> Ref #359

我目前只是粗暴的直接将源码里的字符替换成中文，没有像你一样进行额外的添加，您的建议如何？

**Reply by People-11** [reply]
> @People-11 FYI：
> 
> * 文档仓库：https://github.com/spencerwooo/onedrive-vercel-index-docs
> * ~Next.js 国际化：https://nextjs.org/docs/advanced-features/i18n-routing~ 应该用不到这个，下面的文档就够用了
> * Nextra 文档国际化：https://nextra.vercel.app/features/i18n
> * 中文文档排版指南：https://github.com/sparanoid/chinese-copywriting-guidelines
> 
> edit - 我是不是理解错了，我以为你也要翻译文档，要是我理解错了就忽略我吧

如果您有需要，我可以在闲时进行翻译工作：我不太懂编程，所以我将新开一个fork然后对里面的内容进行翻译

**Reply by myl7** [reply]
> > Ref #359
> 
> 我目前只是粗暴的直接将源码里的字符替换成中文，没有像你一样进行额外的添加，您的建议如何？

如果你有兴趣以及精力的话，可以跟踪一下 #359 这个 PR，其中 `public/locales` 下的 json 文件里就是提取出来的所有需要翻译的字符串，格式是 i18next JSON。给这个 JSON 文件里的所有 key 都填上 value 就翻译完了。

因为需要翻译的字符串不多，我也会给一份翻译。不过可能有不太信达雅的地方，也请你适情景订正。

当然整个翻译工作要开始的话还是要这个 i18n PR reviewed & passed & merged 了之后才能开始。

**Reply by People-11** [reply]
> > > Ref #359
> > 
> > 
> > 我目前只是粗暴的直接将源码里的字符替换成中文，没有像你一样进行额外的添加，您的建议如何？
> 
> 如果你有兴趣以及精力的话，可以跟踪一下 #359 这个 PR，其中 `public/locales` 下的 json 文件里就是提取出来的所有需要翻译的字符串，格式是 i18next JSON。给这个 JSON 文件里的所有 key 都填上 value 就翻译完了。
> 
> 因为需要翻译的字符串不多，我也会给一份翻译。不过可能有不太信达雅的地方，也请你适情景订正。
> 
> 当然整个翻译工作要开始的话还是要这个 i18n PR reviewed & passed & merged 了之后才能开始。

整个项目不止30个地方需要翻译，您可以看看我的 [fork](https://github.com/People-11/onedrive-vercel-index)，或者我的 [部署](https://people11.vercel.app/)，我也不敢保证我进行了100%的翻译

**Reply by myl7** [reply]
> 

确实不只，因为这个 PR 还没 hook 完……

如果你方便的话也可以帮忙检查一下没有 hook 住的地方，那个 PR 下也有 preview。

**Reply by People-11** [reply]
> > 
> 
> 确实不只，因为这个 PR 还没 hook 完……
> 
> 如果你方便的话也可以帮忙检查一下没有 hook 住的地方，那个 PR 下也有 preview。

我看了一下您的翻译，并且按照对此项目使用的理解进行了一点儿修改。但是我不知道如何单独发起 PR ，可否请您看一下我修改后的 [文件](https://github.com/People-11/onedrive-vercel-index/blob/main/public/locales/zh-CN/common.json) ？

**Reply by myl7** [reply]
网上关于如何发起 PR 的教程应该挺多的，可以搜索一篇出来参考一下。
就把修改了的这份文件覆盖了，`git add` `git commit` `git push` 然后在 GitHub 页面上提交就行。

**Comment by Seele0oO** [comment]
所以说，有人需要这个项目的文档翻译嘛？我可以尝试翻译一下？

**Reply by People-11** [reply]
我正在着手文档的翻译，今晚应该能完成一部分

更新：是真的多，但我只剩两个页面了。翻译完成后我会再总体检查一遍，然后 Push 到我的 [Fork](https://github.com/People-11/onedrive-vercel-index-docs/tree/zh_CN) 里。

更新：真的好多啊……我翻译完就去休息了……白天再检查和Push吧……

**Comment by People-11** [comment]
中文文档已翻译完成：[Fork](https://github.com/People-11/onedrive-vercel-index-docs/tree/zh_CN)
不过似乎创建 Pull Request 会覆盖原本的文档，我目前不知道该怎么做。
更新：我想我应该保留原来的文件副本而另外创建翻译后的文件副本。这只是简单的文件重命名工作，为了 Commit 的整洁性，我将会重新 Fork 并且提交新的文件结构。
更新：我发现还有一些细小的地方需要修改，我先重新 Fork ，等我修改妥善后会 Push 的

**Reply by Seele0oO** [reply]
是的，创建pull request然后由上游审核并合并。

**Reply by transflo** [reply]
您好，我也想提供翻译，或许我们可以分一下工

**Reply by People-11** [reply]
> 您好，我也想提供翻译，或许我们可以分一下工

我已经完成翻译了，现在只是在进行检查及收尾工作

**Comment by spencerwooo** [comment]
@People-11 https://nextra.vercel.app/features/i18n

**Reply by People-11** [reply]
我使用了另外的方案（很抱歉我没看到这个教程），这是我的 [PR](https://github.com/spencerwooo/onedrive-vercel-index-docs/pull/2)，也许可以使用这个方案？
更新：看起来不难，只是需要一些工作，我可以关闭 PR 并尝试进行教程中的设置，您的意见如何？
更新：原 PR 已关闭，我已经按照文档进行文件名更改和配置添加，新的 [PR](https://github.com/spencerwooo/onedrive-vercel-index-docs/pull/4)

**Comment by spencerwooo** [comment]
Resolved with https://github.com/spencerwooo/onedrive-vercel-index-docs/pull/4
